crates/pessimistic-proof/src/global_index.rs

```diff
@@ -0,0 +1,89 @@
+use reth_primitives::U256;
+use serde::{Deserialize, Serialize};
+
+/// The [`GlobalIndex`] uniquely references one leaf within one Global Exit Tree.
+/// Further defined by the LXLY specifications.
+/// | 191 bits |    1 bit      |    32 bits   |      32 bits     |
+/// |    0     |  mainnet flag | rollup index | local root index |
+#[derive(Debug, Clone, Serialize, Deserialize, Copy, Default, PartialEq)]
+pub struct GlobalIndex {
+    pub mainnet_flag: bool,
+    pub rollup_index: u32,
+    pub leaf_index: u32,
+}
+
+impl GlobalIndex {
+    const MAINNET_FLAG_OFFSET: usize = 2 * 32;
+}
+
+impl From<U256> for GlobalIndex {
+    fn from(value: U256) -> Self {
+        let bytes = value.as_le_slice();
+
+        let mainnet_flag = value.bit(Self::MAINNET_FLAG_OFFSET);
+        let rollup_index = u32::from_le_bytes(bytes[4..8].try_into().unwrap());
+        let leaf_index = u32::from_le_bytes(bytes[0..4].try_into().unwrap());
+
+        Self {
+            mainnet_flag,
+            rollup_index,
+            leaf_index,
+        }
+    }
+}
+
+pub trait FromIndex {
+    fn from_index(i: GlobalIndex) -> Self;
+}
+
+impl FromIndex for U256 {
```

Comment:
Why is this a trait rather than plain method on `GlobalIndex`? What are the various impls you envision?

Reply:
That results of some trial to implement `impl From<GlobalIndex> for U256` but I ran into issues. 

I removed the custom trait and added directly a method `to_u256` in `GlobalIndex` instead